### ISSUE TITLE
Organism ID vs Entity ID vs Agent

### ISSUE THREAD

**Issue by campmlc** [ArctosDB/arctos#3765]
I just went through the process of creating an entity for the first time, which was excrutiating. I have no idea if I did anything correctly. https://arctos.database.museum/entity.cfm?action=edit&entity_id=4
The process is opaque, and even if documentation were present this is an incredibly complex operation.
After I created the entity, I added the entity IDs (as url - this is not clear) to each record as an organism ID. 
I could then click the entity and see all the records, whew.
Then, I experimented with the same thing by creating an agent for the same animal, "Bernice". She was a chimp at the ABQ zoo.
I added all her IDs as akas to her agent profile. 
I added her as an agent to each record (as "subject").
Much easier, much clearer, no need for reciprocally connecting anything.
And voila, all the specimen records and identifiers show up in a single, clear dashboard on her agent page. 
I strongly suggest, yet again, that we incorporate the agent model for use with organisms. It works and delivers precisely what we need.

**Comment by campmlc** [comment]
To minimize confusion, changed preferred name to "Bernice Pan troglodytes" to distinguish from any other Bernices out there.

**Comment by Jegelewicz** [comment]
I changed Bernice's agent type to other agent.

**Comment by Jegelewicz** [comment]
Here is a [project for Bernice](https://arctos.database.museum/project/10003584).

What can it do that an agent can't?

- no birth death date (unless we use start and end for that, which isn't explicit, but sorta works)
- no relationships to parents, siblings,  children (these could be manually created as related projects)
- Accessions for an organism would need to include ONLY that organism, probably NOT realistic
- other identifiers need to be added to every single catalog record manually

What can and Agent do that a project can't?
- explicit birth and death dates
- explicit relationships to other agents
- explicit identifiers as akas, no need to add them to every single catalog record
- addresses (zoo and experimental animals move around)


What other ways should we evaluate this idea?

**Comment by campmlc** [comment]
We discussed yesterday in intern meeting - (I also changed type to other agent - not sure why it didn't save).
@mkoo @KyndallH

**Comment by Jegelewicz** [comment]
I had to remove "first name" as other agents cannot have a first name.

**Comment by Jegelewicz** [comment]
@dustymc has always said we should just catalog organisms. We had some discussion about organisms during the TDWG MaterialSample Task Group meeting yesterday. In order to create an organism ID GUID, the Field museum creates a new "specimen-less" catalog record. Given this and all of our discussions, here is what I propose, because I think we can do slightly better for our community.

1. We create a community managed collection. (Arctos:Entity)
2. Arctos:Entity will use the Teach collection code
3. Arctos:Entity will be managed by those with manage_code_table and requests for new entities will be handled the same way code table requests are
4. Records in Arctos:Entity will ALWAYS be part-less
5. GUIDS from Arctos:Entity can be used as values in other identifier = Organism ID
6. We create a few new event types: birth and death so that we can record these events for organisms when they are known

The reason I propose this special collection for organisms is that it will help prevent the creation of duplicate identifiers in Arctos for any given organism. It also allows us to share the burden of keeping up with them and will not impose additional fees for collections doing the work to connect things. I have settled on "Entity" because cultural collections may have use for this as well - to bring together various parts of a set and so on.

Over time, I think we will find opportunities to seed catalog records from Arctos:Entity and to add data to Arctos:Entity from other catalog records, but we can work those out as they become apparent and useful. We will likely need some rules about when an Arctos:Entity record should be created, but I think they can be fairly simple.

What about this is nuts? I'm sure something is!

**Comment by campmlc** [comment]
This is definitely an interesting idea and corresponds with previous
suggestions for a synthetic "organism view" that would combine multiple
records from the same individual via the UI. This makes more sense than
that approach. How would we integrate this with other platforms and GBIF?

**Comment by Jegelewicz** [comment]
>  How would we integrate this with other platforms and GBIF?

By sending the url for the Arctos:Entity record as the Organism ID for the catalog record item. This is exactly what Field Museum does. We would NOT transmit anything in the Arctos:Entity collection to GBIF (yet). Eventually, we may want to send that information in its own kind of Darwin Core Archive, but for now, it would remain with us (although everyone could see it through the Organism ID link).

**Comment by dustymc** [comment]
>@dustymc has always said w

No, the data themselves are and have been doing that. (I suppose it's too late to switch my title from "data janitor" to "Speaker for the Data"?)

>community managed collection

I have been saying bigger is better - something bigger than Arctos would be better, but if we must do this then this seems as good as it gets.

>managed by those with manage_code_table

This seems entirely unnecessary, I'm not sure what the goals of such restrictions would be, but it's also "details" that can be easily adjusted at any time using familiar tools.

>ALWAYS be part-less

I don't get that either (but it's also just more details). Why not accumulate parts? "This critter has blood samples in these 48 places that we know of, none of them have useful public data" seems incredibly useful, condition ("we have no idea") and disposition ("not here") handle the details.

>GUIDS from Arctos:Entity can be used as values in other identifier = Organism ID

and

>"Entity" because cultural collections

1. Yes, a GUID in the whatever-its-called collection is just an identifier capable of carrying the kinds of data that keep being mentioned, the collection needs a name but the identifier itself can be used for WHATEVER. "Organism ID" is one of those.
1. Yep, "a whole bunch of people ran off with chunks of this thing" can't be too rare. (But maybe any of those that anyone cares about have inherent names - I've got a chunk of The Berlin Wall, it wasn't the only wall around and I don't have any fancy numbers scribbled on it but everybody knows what it is anyway.)

>few new event types

Attributes - events are interactions with humans, Attributes are - well, Attributes. (I was wondering how we don't have this, turns out we do but we call it "numeric age" and  the collecting event after "verbatim preservation date." Might be a good opportunity for some reconciliation.)

>prevent the creation of duplicate identifiers in Arctos for any given organism

I think that's an impossible (albeit worthy) goal. The printer stuck, 12 XYZ pages got printed, now there are 273 things that say "XYZ123" on them out there. Some sort of filter seems very useful, but starting out with the idea that you can do something that you won't actually be able to do will be frustrating. Duplicates WILL be created, and this data object is capable of doing something about it: https://handbook.arctosdb.org/documentation/catalog.html#recataloging-specimens

>share the burden

Not if doing so requires manage_codetables. This would just be a collection, you can grant access to anyone who can demonstrate that they understand how to use it. 

>need some rules

I'll again advocate for openness/inclusiveness. If the bar is very high then this just won't get used. Definitely rules and guidelines are needed, but I think they should generally encourage new "entities" when there's some question - "this seems like something that might have a chunk cataloged elsewhere" would be a HUGE benefit for some future researcher who might be willing to dig around in the collection and maybe make everyone's data better.

Anyway, all details, I still don't see a better approach or more appropriate data object for this.

>How would we integrate this with other platforms

Such as?

>and GBIF

AFAIK they don't accept Organisms but they do Occurrence-->OrganismID - we'd just give them what they can handle.

**Comment by Jegelewicz** [comment]
I have submitted a prospective collection request so we can discuss there.

**Comment by dustymc** [comment]
>discuss there

I don't know where "there" is - should I?

**Comment by Jegelewicz** [comment]
> I don't know where "there" is - should I?

It will become an issue in the New Collections repo, which is where we make decisions about incoming stuff...

**Comment by ewommack** [comment]
AWG at 9 Dec 2021 meeting agreed to try it out.

**Comment by Jegelewicz** [comment]
Please review the project - https://github.com/ArctosDB/new-collections/projects/59

**Comment by ccicero** [comment]
@Jegelewicz @dustymc @campmlc 

I don't need if this is the right place to post this, but I just finished uploading MVZObs:Bird records that go with MVZ:Bird records (= same organisms = same entity) - with reciprocal relationships 'same individual as'

Here is the [saved search](https://arctos.database.museum/saved/DougBell_GoldenEagles). 

This will be a good test case once the new 'entity' collection is set up.

**Comment by campmlc** [comment]
Will you be adding additional records for the ones that died,
e.g. MVZ:Bird:193195?
For that one, you have one record with an encounter event, another record
cataloged in your Obs catalog, and a final record for the carcass?

**Comment by Jegelewicz** [comment]
@dustymc @campmlc @ccicero 

The Arctos Entity Collection is live! Before we start writing up procedures and such, I'd like to create a couple of entities so that we can review and talk about how it should work for the community.

I was going to test with Kianga and then one pair of Carla's bird/observation records.

Is everyone OK with me doing that?

**Comment by Jegelewicz** [comment]
MVZ Bird test
https://arctos.database.museum/guid/MVZ:Bird:193195
https://arctos.database.museum/guid/MVZObs:Bird:4777

**Comment by campmlc** [comment]
Yes

**Comment by ccicero** [comment]
Awesome Teresa, and go for it. Thanks!

Mariel - with regard to your question, we only will be getting the blood samples, no carcasses (even if the bird died). So the data in Arctos are what we have, and I don't expect more for those records.

**Comment by Jegelewicz** [comment]
OK - [Kianga](https://arctos.database.museum/guid/Arctos:Entity:1) has been created in the Arctos:Entity collection.

A few observations:

1. When I put the link for the entity record (https://arctos.database.museum/guid/Arctos:Entity:1) in Organism ID, it does not show up on the catalog record.

So this
![image](https://user-images.githubusercontent.com/5725767/149221210-24d6b6d9-bcfc-4281-b953-e2f2c5a33191.png)

Looks like this in the catalog record
![image](https://user-images.githubusercontent.com/5725767/149221289-305ad11e-3dab-4dc2-b2b3-7c3e4e1ca238.png)

2. We need to think about how the information in the entity should propagate to the catalog records with parts from it/observations of it.

So that all of this which is in the entity
![image](https://user-images.githubusercontent.com/5725767/149221464-22cebe4c-6f11-4485-8b0b-60c95c2cf322.png)

will be complete in all of the catalog records of things from that entity
![image](https://user-images.githubusercontent.com/5725767/149221547-5b7a7675-0286-4716-b75a-53a08ec3b663.png)

3. Some stuff in the entity records needs more thoughtful consideration. For Kianga, birth and death might be good events to include in the entity record while things like "collection" and "observation" would be more useful in the object/observation records.

4. I used "living specimen" as the record type for Kianga, but I wonder if "individual" and "assemblage" might be better types for this collection. Kianga will eventually die, but the record in the entity collection will never hold a "specimen" so I think we need to decide how to handle it.

5. We should also consider how entity attributes populate/are populated by attributes in the object/observation records.
![image](https://user-images.githubusercontent.com/5725767/149222503-3ec4976b-2092-4ad6-a6d0-58ff3e25a55f.png)
vs
![image](https://user-images.githubusercontent.com/5725767/149222552-ef0c9d82-d2ee-4faf-8511-aa861fcf6a48.png)

And there is probably more, but I think a focused discussion group will be helpful.

If everyone will complete this [When2Meet](https://www.when2meet.com/?14133248-91bfL) maybe we can meet up for an hour and talk about it?

@campmlc @ccicero @dustymc anyone else?

**Comment by dustymc** [comment]
>When I put the link for the entity record (https://arctos.database.museum/guid/Arctos:Entity:1) in Organism ID, it does not show up on the catalog record.

I don't think you added it, but I don't have access so IDK. There is some magic around OrganismID (or subsets thereof, or something) that will need redone, that could be the problem.

>how the information in the entity should propagate to the catalog records 

I'm certainly up for discussion, but I don't think it should (or can). Propagating the other way - pulling stuff to the entity - seems more useful/safer.

>will be complete in all of the catalog records of things from that entity

I think that's just wrong and confusing. THIS blood sample says "123" - adding "456" (some other sample from some other time/place/collection/whatever) to that record just adds confusion and results in nobody quite knowing what they're looking at. The entity could/should bear both "123" and "456" but I think the other way - at least via automation - is probably just evil.


"living specimen" and "individual" sound like same thing to me, at least for mammals. "assemblage" seems just wrong - we're not building a critter from the samples (yet?!)....

>for this collection

I don't think the collection will be that homogeneous, at least if anyone uses it. It's the proper place to catalog an "assemblage" of dinnerware, so the set can be pieced back together from the 42 pieces (and photos and correspondence and ....) scattered across 20 museums, for example. Also pack of wolves, population of sponges, whatever - anything that seems useful as a THING to someone for some reason.

**Comment by ccicero** [comment]
An update on my first entity project (yay!), for the record:

Here is the [project](https://arctos.database.museum/project/10003566) on the Golden Eagle tracking. I bulkloaded the entity records and then bulkoaded relationships to MVZ:Bird and MVZObs:Bird.

Following our discussion today, I decided that a general locality/event is better than no locality/event. This way one can see the range of dates and general region where eagles were followed with radio telemetry after banding and sampling for blood. 

I used the existing locality ID of 'San Francisco Bay Region' for all birds that stayed within the region (but maybe were tracked in different counties), and entered a broader range for ones that travelled farther (e.g., 'San Francisco Bay Region to Mexico."

For verbatim locality, I put 'Entity locality - see individual records for specific localities' ; for the dates, I put the range of dates from the first trapping to the date of the last known position.

In remarks I put 'Bird captured, banded, sampled for blood, and followed using radio telemetry. See individual records for details of specific occurrences.'

It is a bit of a pain to get the data together, but that's a function of how we receive the data. The Arctos part of the process is fairly straightforward. I have another batch of >300 blood samples, some multiple samples from the same individual. I think (hope) those data will be easier to work with.

**Comment by dustymc** [comment]
It's not clear what the entity is doing in this situation - it's just one more relationship, something we've always been able to do. Suggesting adding the EntityID (https://arctos.database.museum/guid/Arctos:Entity:10) to the "real" records (eg https://arctos.database.museum/guid/MVZ:Bird:193206), probably as type [Organism ID](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type#organism_id). That won't be very interesting within Arctos -  that's not the point of "entities" anyway - but that will provide a unique, resolvable identifier for the "components" in eg GBIF, which is new and interesting.

**Comment by Jegelewicz** [comment]
What @dustymc said - these should be Organism ID in the catalog records - my bad for not making that clear when we talked about it yesterday, but easy enough to add with the other identifier bulkloader.

Now that we have some great examples to look at - it is time to schedule a focus group to talk about how this could work better. One of the questions is what should the "reciprocal" of Organism ID be on the entity record? One of the goals is to get at all of the catalog records for things from this entity and right now that doesn't happen.

**Comment by ccicero** [comment]
@dustymc @Jegelewicz Why is it Organism ID and not Arctos:Entity ?

If I use Organism ID, what do I enter?

https://arctos.database.museum/guid/MVZ:Bird:193196 ---> https://arctos.database.museum/guid/Arctos:Entity:3
https://arctos.database.museum/guid/MVZObs:Bird:4778 ---> https://arctos.database.museum/guid/Arctos:Entity:3

???

**Comment by dustymc** [comment]
>Why is it Organism ID and not Arctos:Entity ?

https://dwc.tdwg.org/list/#dwc_organismID (those links never work, search for organismID)

"Entity" is very general, can be used for lots of things (most of which we presumably haven't thought of yet).

"Organism" is (sorta) specific in DWC-land.

"Organism" is one type of "entity" in Arctos; maybe we'll eventually be mapping dinnerware sets "grouped" by Arctos:Entity to some field in some not-DWC exchange standard.


>what do I enter

ID Type=``Organism ID`` value=``https://arctos.database.museum/guid/Arctos:Entity:3`` on eg https://arctos.database.museum/guid/MVZ:Bird:193196

**Comment by campmlc** [comment]
You could include the relationship "same individual as" in your identifier, although this isn't strictly necessary if Dusty creates a link out. We should decide on this. It makes sense to me to have the organism ID display under relationships.

**Comment by dustymc** [comment]
> "same individual as"

I'd probably call that "just wrong" - at least in this case, the individual isn't what's being cataloged. (In the case of two collections cataloging bits of the same dead rat it would be, so maybe there is some room/reason to clarify through relationships.)

>Dusty creates a link out.

Hu?

>have the organism ID display under relationships

The Organism ID is an attribute of each of the pieces. The "components" are not the 'same individual,' they're a piece of the individual. 

Also of note: the Arctos:Entity is but one thing which can be used as OrganismID. Whatever's done as Organism ID with Arctos:Entity should also work for things like band numbers and some future glorious OrganismBank.

**Comment by ccicero** [comment]
I understand that an entity can be different things, organismal or not. But I'm still a bit confused. Are you all saying that I should select 'Organism ID' as the identifier, but then enter the URL for the Entity ID? Is this mostly so that the Entity ID gets included in the DwC Archive under Organism ID for data aggregators?

It is separate occurrences of the same individual, so why not choose 'same individual as' ? What's the alternative?

**Comment by campmlc** [comment]
select 'Organism ID' as the identifier, but then enter the URL for the
Entity ID? Yes
Is this mostly so that the Entity ID gets included in the DwC Archive under
Organism ID for data aggregators? - Yes
And my view is use same individual as. The alternative is self.

**Comment by dustymc** [comment]
> select 'Organism ID' as the identifier, but then enter the URL for the Entity ID

Yes - and please let me know when you've done that to a record so I can make sure flat and DWC are working as they should.

I think the simplest and most consistent thing is to not make relationships to or from the entity at all. Some kinds of OrganismIDs won't be able to support that anyway, and within Arctos the entity doesn't *do* anything new, other than provide a (very nice!) shared identifier. I just can't see any actual benefit to adding relationships - those get at my comments above - and there is value in "simple" and "consistent." 

You should definitely make relationships between the bits when possible, but those do not need to involve the entity.

**Comment by campmlc** [comment]
I still think we should make the relationships explicit, because this makes
sense to humans. Including relationships also provides a link to the "Find
all" and "Find all same individual as" tools which are highly useful, if
obscure. The alternative "search" tool next to the identifiers which lack
relationships is even more obscure.

**Comment by campmlc** [comment]
And we can't make relationships between the bits, if the bits are guids,
and if there are more than three of them. This is the reason we need the
entity in the first place - too many reciprocals to keep track of.

**Comment by dustymc** [comment]
>too many reciprocals to keep track of.

And the proposed solution is to add a multiplier?!

Unrelated to Organisms, https://github.com/ArctosDB/arctos/issues/4101 offers a mechanism by which reciprocals (and other derived assertions) might take care of themselves.

**Comment by Jegelewicz** [comment]
Note that when I put the entity url in Organism ID with relationship of "self" - it just doesn't show up in the catalog record....

![image](https://user-images.githubusercontent.com/5725767/152588409-91efed4b-37cd-44b1-8a14-7aad70d2caa3.png)

gets me this

![image](https://user-images.githubusercontent.com/5725767/152588492-2098292b-4256-42d9-ae5e-e268c61f0ffa.png)

The only way to see the Organism ID is to edit other identifiers....

**Comment by Jegelewicz** [comment]
https://arctos.database.museum/guid/MSB:Mamm:326433

**Comment by Jegelewicz** [comment]
vs. https://arctos.database.museum/guid/MSB:Mamm:326441

**Comment by dustymc** [comment]
>doesn't show up

It will in next release.

**Comment by Jegelewicz** [comment]
Cool. But then the issue remains - when I am on the entity page, how can I easily get to all of the catalog records that are "part" of that entity?

**Comment by ccicero** [comment]
So I'm hearing conflicting things:

1) Don't make relationships (Dusty) vs Yes to relationships (Mariel, Teresa). I agree that making relationships makes sense to easily get at all related catalog records.
2) If we do make relationships, use "same individual as" (Mariel, Teresa?) or "self" (Dusty?) - 'same individual as' is more explicit

???

**Comment by Jegelewicz** [comment]
I agree with Dusty - use Organism ID and self and weirdly, this should also be added to the entity record - see https://arctos.database.museum/guid/Arctos:Entity:1

**Comment by dustymc** [comment]
>how can I easily get to all of the catalog records that are "part" of that entity?

Give the entity it's own ID, click 'search'?

<img width="586" alt="Screen Shot 2022-02-04 at 11 26 34 AM" src="https://user-images.githubusercontent.com/5720791/152590839-a0c2d386-b727-4799-91b3-a9e35677c0ff.png">

>If we do make relationships,

There are, say, three THINGs involved:

1. Chunk of bird
2. Chunk of bird
3. abstract entity representing the bird as as an organism

I don't think it makes sense that the relationship between (1) and (2) is the same as the relationship between (1) and (3), and as far as I can tell that's what's being offered.

I don't think relationships are *wrong*, just that they're not really necessary and aren't possible on all types of Organism IDs. I don't think it can hurt anything to add them, I'm just not sure that's complexity worth "requiring" (via documentation - we probably can't actually require anything).

**Comment by Jegelewicz** [comment]
You beat me to it!

![image](https://user-images.githubusercontent.com/5725767/152591889-93383272-c935-41c7-af47-0942c8b124b6.png)

on ANY of the records with that identifier finds them all

![image](https://user-images.githubusercontent.com/5725767/152591995-54fb15b1-cfaf-410c-b118-2afc671b49dd.png)

**Comment by Jegelewicz** [comment]
But we need to get rid of the "not found" - what's that doing?

**Comment by dustymc** [comment]
>beat me to it!

About time!

Note that this also works with things like band numbers, or anything else someone might want to use to build an organism (but not from the "entity" which wouldn't exist in that case). Consistency==good.

>rid of the "not found" 

That's part of what's hiding things, gone in next release.

**Comment by campmlc** [comment]
Then we need a tool to " find all same organism as" somewhere on the
catalog record page and on the entity page.

**Comment by campmlc** [comment]
But that is obscure. Need  what I just posted.

**Comment by Jegelewicz** [comment]
I think it would be good to change "search" to "find all" - maybe that would make it more obvious?

**Comment by Jegelewicz** [comment]
@dustymc @ccicero @campmlc 

Let's set a meeting to do some entity/organism planning - https://www.when2meet.com/?14465690-n8yUW 

I'd like to make this a workshop - maybe spend 2 or 3 hours hashing through some stuff so please look for a time in your schedule when you can spare a few hours. If there is anyone else who should be there, please tag them. I will check the when2meet late next week and then send a calendar invite. Thanks!

**Comment by ccicero** [comment]
Yes please! I'm getting lost in this discussion and still not sure of the best thing to do. For now I'll do nothing to my records until we get this sorted out.

I just filled out the poll. My preference is for a Thursday if possible. Mondays I can do but not until 2/28.

Something funky has happened. Arctos:Entity doesn't show up anymore in the collection dropdown. And Arctos:Entity:2 was a cultural object but now it's an eagle.

![image](https://user-images.githubusercontent.com/967275/152611575-f1aa7684-e70b-44d7-aeb9-adc064137a25.png)

https://arctos.database.museum/guid/Arctos:Entity:2

**Comment by dustymc** [comment]
Thursdays are good for me too.

>not sure of the best thing to do

Get some sort of an identifier and enter it as Organism ID on your "bits-n-pieces" records. That is all that's necessary to serve the core functionality.

``https://arctos.database.museum/guid/Arctos:Entity:2`` is nice for "some sort of an identifier" because it's resolvable and capable of carrying an immense amount of information and all that jazz, but that's all extra, and that information, along with relationships or WHATEVER, can be added (perhaps even by automation) at any time. Making some sort of record is all that's really necessary. 

("Necessary" isn't correct either, things like USWFS band numbers are fine for Organism ID as well, but of course come with less capability.)

>Something funky has happened. 

I don't have access to that collection and I'm not sure I can much help without.

**Comment by campmlc** [comment]
@Jegelewicz can you give @campmlc @ccicero and @dustymc access to the Entity collection?

**Comment by Jegelewicz** [comment]
Carla already had access - Mariel and Dusty granted.

**Comment by Jegelewicz** [comment]
I don't know what Carla is seeing, but https://arctos.database.museum/guid/Arctos:Entity:2 is still a dress...

**Comment by dustymc** [comment]
Thx, that helps!

Carla maybe you're looking at Attempt One, https://arctos.database.museum/entity.cfm - that needs removed, but we can make redirects to handle that in the new version if necessary before it gets completely nuked.

Display stuff is in production.

**Comment by ccicero** [comment]
This is so weird! I click on the link Teresa sent and #2 is an eagle for me - with old data. 

![image](https://user-images.githubusercontent.com/967275/152658172-399d9caa-4fe4-4678-be74-08e71c07e73b.png)

It wasn't an eagle before. And I could previously search on Arctos:Entity in the collection dropdown but that's gone now.

I've also tried different browsers.

**Comment by dustymc** [comment]
AHA! You're not signed in; something got sideways in the cache, I see the same thing. I just set it all to refresh.

Signed in and out both work for me, if we're talking about the same collection dropdown.

<img width="1426" alt="Screen Shot 2022-02-05 at 1 08 49 PM" src="https://user-images.githubusercontent.com/5720791/152659125-6ee938b5-2130-43b1-b470-d94b84649ab9.png">

**Comment by ccicero** [comment]
Aha! When signed in, Arctos:Entity appears as the first collection:

![image](https://user-images.githubusercontent.com/967275/152660175-cf2e976e-2117-451b-b925-70a553774598.png)

When not signed in, Arctos:Entity appears alphabetically (I wasn't looking for it there):

![image](https://user-images.githubusercontent.com/967275/152660200-e23907e7-7aae-463e-a0fb-fd42516b0a0c.png)

Can we make that show up at the top too?

Either way, the eagle is now back to a dress :)

**Comment by ccicero** [comment]
Never mind, I get it. When signed in I don't have access to all collections which is why it's showing at the top. 

I like it at the top. Can we force it there? Seems lost otherwise.

**Comment by campmlc** [comment]
Agree

**Comment by dustymc** [comment]
I don't think I'll be convinced that non-ordered values are good UI.

Highlighting a tiny collection that by definition doesn't hold anything real of its own doesn't make much sense to me either.

I don't think there are technical barriers, could be added to AWG discussion.

**Comment by ccicero** [comment]
To me it's not equivalent to searching on other Arctos collections because it's not a 'real' collection. If we want it discoverable, we need to make it more visible.

Let's discuss options when we meet.

**Comment by ccicero** [comment]
@dustymc I'm getting an error 'record not found' for the MVZObs:Bird records for which I'm trying to add an Organism ID. But the records are there. e.g. [https://arctos.database.museum/guid/MVZObs:Bird:4780](https://arctos.database.museum/guid/MVZObs:Bird:4780)

It loaded fine for MVZ:Bird but none of the MVZObs:Bird loaded - ???

![image](https://user-images.githubusercontent.com/967275/156111918-610e0c84-ab16-4ac0-90d3-cf16f540e981.png)

**Comment by dustymc** [comment]
You had an extra space or something, I fixed, would not happen under https://github.com/ArctosDB/arctos/issues/4200

``` 
 : MVZObs:Bird:4780:
 : MVZObs:Bird:4781:
 : MVZObs:Bird:4782:
 : MVZObs:Bird:4783:
 : MVZObs:Bird:4785:
 : MVZObs:Bird:4796:
 : MVZObs:Bird:4797:
 : MVZObs:Bird:4802:
 : MVZObs:Bird:4803:
 : MVZObs:Bird:4784:
 : MVZObs:Bird:4778:
 : MVZObs:Bird:4810:
 : MVZObs:Bird:4811:
 : MVZObs:Bird:4779:
 : MVZObs:Bird:4807:
 : MVZObs:Bird:4808:
 : MVZObs:Bird:4809:
 : MVZObs:Bird:4786:
 : MVZObs:Bird:4787:
 : MVZObs:Bird:4788:
 : MVZObs:Bird:4789:
 : MVZObs:Bird:4790:
 : MVZObs:Bird:4791:
 : MVZObs:Bird:4792:
 : MVZObs:Bird:4793:
 : MVZObs:Bird:4794:
 : MVZObs:Bird:4795:
 : MVZObs:Bird:4798:
 : MVZObs:Bird:4799:
 : MVZObs:Bird:4800:
 : MVZObs:Bird:4801:
 : MVZObs:Bird:4804:
 : MVZObs:Bird:4805:
 : MVZObs:Bird:4806:

```

**Comment by ccicero** [comment]
Duh, thanks!

Related issue: When you click on the 'search' link next to the entity in the Organism ID:

![image](https://user-images.githubusercontent.com/967275/156213998-c5b1a854-dc1d-4d9e-8e71-4625d9038fd7.png)

You get 'like' results, so not just Arctos:Entity:3 but also others that contain the number 3 (30 and 31):

![image](https://user-images.githubusercontent.com/967275/156214187-bc1d7d09-f16e-4049-8822-3d94bc71a0a8.png)

Seems  like that should be an exact match.

**Comment by dustymc** [comment]
>Seems like that should be an exact match.

Thx, agreed, in next release.

**Comment by ccicero** [comment]
@dustymc @Jegelewicz @campmlc Following up on our meeting today, I'm trying to create a new Arctos:Entity record for Dusty to try his magic. I'm trying it in data entry (just filling in the required fields). Some basic procedural questions:

1) Event determiner - should that be me or the collector? I'm determining that it's an entity, but the data are coming from the collector.

2) Event determination date - related question: should that be today's date (if I'm the determiner, then yes)? The entity is comprised of different dates. If the determiner is the collector, then what date to use?

3) Agents - original collector?

4) Data entry requires a collecting event, which defeats the purpose of the magic. I can bulkload the entity records but wanted to try this just on one record so Dusty can test out his magic.

Here are the records. I have not yet created the relationships 'same individual as'
https://arctos.database.museum/guid/MVZ:Bird:186456
https://arctos.database.museum/guid/MVZ:Bird:186457

Given that these have the same identifier (USGS Bird Banding Laboratory (BBL): 1381-79365) is there a way to do even more magic by searching for records with the same identifier and magic-ing those somehow into new entities (or using relationships "same individual as")?  How can we make it easier to create the entities where the records are within Arctos?

**Comment by dustymc** [comment]
If it's one record, you can use the clone-->entity widget.

>not yet created the relationships 'same individual as'

I can do that if we can get https://github.com/ArctosDB/arctos/issues/4101 resolved.

>searching for records with the same identifier and magic-ing those somehow

I think that would be a one-time thing, but yes - but I think after we get a few examples all happy and make sure we all like whatever we end up with.

**Comment by ccicero** [comment]
Sorry, how do you get to the clone widget? Is it from 'Other IDs' ---> 'Clone this record' 
?

**Comment by ccicero** [comment]
OK, I'll give it a try!

**Comment by dustymc** [comment]
http://test.arctos.database.museum/guid/Arctos:Entity:5

<img width="1618" alt="Screen Shot 2022-03-17 at 4 07 01 PM" src="https://user-images.githubusercontent.com/5720791/158908429-4fc91732-48d2-479f-b477-e0d7e3c8dc7d.png">

Unless I break something important I'll try to push that out tonight or tomorrow. It's set up so that the entity pane is pretty easy to change, if you want to see something else there. So with that the recipe would be

* create an Entity, potentially with very minimal data
* use the ID as OrganismID (or whatever, it's currently just looking for the identifier however it's used, also easy to change)
* do SOMETHING with https://github.com/ArctosDB/arctos/issues/4101 ("add asserted_by and asserted_date and ignore the rest for lack of interest" would do what I need) and I'll magic other stuff, which wouldn't add any information but could save some clicks.

**Comment by ccicero** [comment]
A little late, but I like it!  

Some thoughts:

I suggest removing the 'parts' block because entities shouldn't have their own parts and the parts are shown in the components summary. It's distracting and a bit confusing to have the parts block show but with no parts, although there are parts in the summary.

I think it makes sense to remove the relationships since they're not really relationships but component parts as in the summary.

I think it's confusing to have it show 0 events (as in your test example), but then have events show in the summary. Rather than show this:

![image](https://user-images.githubusercontent.com/967275/159373504-50523d9d-0693-4444-bbf5-f441c3e4bc39.png)

why not put the composite map there with the number of events pulled from the components, and then just list the components below? So it would look something like this:

![image](https://user-images.githubusercontent.com/967275/159373912-5b6b5d40-d8dc-4a97-84a5-58a21ae71fe7.png)

with the component details below like you have them  (but without the map).

Can you try that?

**Comment by dustymc** [comment]
The top of the page is the normal page that gets used for all catalog records, plus one 'if entity' clause near the bottom.

I'm not opposed to something different - either replacing the page with something else altogether, or wrapping a bunch more "if entity..." code around things (if we can afford that), or WHATEVER, but I think more needs worked out before we get too crazy with this. 

>removing the 'parts' block 

We've been back and forth on "what should we do when there's nothing?" a few times, that could be another more general conversation. Maybe it makes sense to just show nothing when there are no (events, parts, whatever), or maybe that looks broken to naive users and some form of "this page intentionally left blank" makes more sense. Or maybe it does make sense to do one thing with entities and something else with other stuff, but I'm not yet convinced.

>because entities shouldn't have their own parts 

I'm not entirely convinced of this either. It seems there are decent arguments on both sides, and those arguments may or may not be the same for "entity components" which are Arctos records vs. those which aren't. (Eg if someone somehow figures out that SomeRandomCollection has a sample listed in their card catalog "database," this would provide a place to record that, maybe along with the fax number you could use to talk to them about it.) 

>although there are parts in the summary.

Sometimes.... Maybe CollectionA uses disposition to flag unavailable parts (yay them!) and CollectionZ just deletes them (not what I'd call ideal, but not something we can control either). Maybe it does or will make sense to try to grab that blood sample as an "entity part" before CollectionZ can get all clicky. (I'm **not** making that argument, but I'm not preemptively refuting it either.)

>remove the relationships

That should be discussed after https://github.com/ArctosDB/arctos/issues/4101 (feel free to make that happen somehow...). I think I currently agree - I might use the Entity to make relationships between components, but I don't think relationships to and from the entity record add anything other than confusion.

**Comment by dustymc** [comment]
AWG: flip the scren

map

component summary list

"directly-attached entity-stuff follows...."

{normal specimen_detail page}

**Comment by mkoo** [comment]
Let me reiterate how amazing a Entity record comprised of multiple collection records is!

**Comment by dustymc** [comment]
Screen-flipping (and some explanatory/layout text) in next release - see http://test.arctos.database.museum/guid/Arctos:Entity:5

<img width="1629" alt="Screen Shot 2022-03-25 at 8 53 40 AM" src="https://user-images.githubusercontent.com/5720791/160155772-71e5248a-c78d-4caa-bb18-e730a25f4498.png">

**Comment by dustymc** [comment]
above table near Entity Components Summary - "find all Arctos records" button

**Comment by Jegelewicz** [comment]
Directly asserted data - don’t display if there is nothing, change to “Entity data not associated with individual Arctos catalog records.”

Entity Components Summary - change to “Arctos records related to this entity.”

**Comment by Jegelewicz** [comment]
- [ ] Add identifiers to the entity magic

**Comment by dustymc** [comment]
>Entity Components Summary - change to “Arctos records related to this entity.”

I didn't do this, for two reasons

1. This isn't necessarily limited to Arctos. I don't think we need to get too hung up on the theoretical, but I'd rather not paint ourselves into any corners we can easily avoid either.
2. "related to" seems overly waffly. This is a much more specific kind of relationship, and one which doesn't use the typical "relationship" mechanism so the term seems worth avoiding. There are lots of ways in which these data can be related, but they're all some kind of components of some kind of entity. 

"change to “Entity data not associated with individual Arctos catalog records.”" and "Add identifiers to the entity magic" are self-conflicting. Even if we don't add tools, some of the entity-stuff will also be asserted, etc., etc., etc. Again I don't think this is more clear than the existing verbiage, so I didn't change anything.

>don’t display if there is nothing,

https://github.com/ArctosDB/arctos/issues/4483

**Comment by ccicero** [comment]
@dustymc @Jegelewicz @campmlc 

I just created a new [entity record](https://arctos.database.museum/guid/Arctos:Entity:33) by cloning a record (Identifiers --->  Clone this record ---> Create an Entity (option 3). I then manually added the Organism ID to the two records in order for the data to get pulled into the entity.

This is a test to resolve some issues before I bulkload the rest. 

**Questions/issues:**

- It created the entity under the MVZ:Bird accession rather than an Arctos:Entity accession which is what we did for the other entities. Do we want entities to have their own accession or be part of the accession of the components? In some ways it makes sense to have it be part of the components accession.

- Even for entities, I think it's valuable to have basic information (who, what, when, where) in the results set if we are making entities searchable (otherwise, you have to click on each record to see the basic data). 
Who - agent
What - identification
Where - HG + spec loc
When - verbatim date

Right now it's blank except for taxon and accession. 
![image](https://user-images.githubusercontent.com/967275/161134584-886905f6-c7a6-49d7-8e96-fdf1068d3041.png)

If people are searching entities, they should get basic information so what can/should be pulled in by Dusty for the results set? And what if there are multiple agents and/or different locations/dates - pull the first one? What about showing the number of events under the entity catnum, like we do for other records with >1 event? 

Or do we even want entities searchable in the dropdown with other collections?

**I'm gone for our next scheduled meeting (in Mexico on vacation April 7-16), so can we meet Monday April 4th instead (10-11am Pacific) to discuss?**

**Comment by campmlc** [comment]
Agree that accessions are a problem. We have entities with dozens of
accessions. One of the problems with the multiple events model was that we
couldn't record more than one accession associated with a record, so
accession info associated with the different parts and events ended up in
remarks. Not functional.  I do agree that we should somehow capture
accessions as related to each sampling/collecting event.

**Comment by dustymc** [comment]
Not completely Entity-related, but your data currently assert that you killed the same bluebird twice. Unless there's CPR involved, you need a better [event type](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_event_type).

>Do we want entities to have their own accession 

At some level I don't care, accessions are administrative and you can do whatever you want with them, but I think accessions dedicated to entities is probably more correct/less confusing. You didn't actually receive anything for Entities, they are by definition not THINGS, you magicked them out of other data, keeping that separated seems worthwhile.

My current inclination is to not attach anything that you don't absolutely have to to the Entities, which will keep them out of most search results. MAYBE that will evolve as we get deeper into this, but if we get there it will be easier to add than to subtract.

We could test excluding them from search results by default, but that's going to have some computational cost; I'm not sure it's worth tackling that until there's some actual reason (eg clearly confused user) to do so.

I could pull whatever data into Entities, but it will get stale (a collector will contact you with a fix, you'll update the "real" catalog record and skip the Entity) so if we do go there I'd like to have some mechanism to tell what's been added by scripts and what might have been added by people for whatever reason. (The current 'entity magic' tab is 'by people,' it just saves you some clicks.)

The 4th at 10 works for me.

**Comment by dustymc** [comment]
>which ... should we pick?

Interesting point. Maybe one of those bluebirds was in Nebraska last week (and maybe you'll even eventually get those data) - semi or implicitly asserting that it wasn't in Nebraska (by adding one or more "California" events) is trending towards "wrong," where asserting (via association) 'we just know this stuff, and we don't know what we don't know!' is a lot easier to defend. Entities should represent something more than the sum of the bits we happen to have some knowledge of, and it seems we might be unnecessarily blurring that line by adding arbitrary stuff from their components.

>and the other identifiers 

One thing I'd like to somehow avoid is asserting self-conflicting messes that will later be difficult to extract ourselves from. If 12345 says "Entity1" and Entity1 says "12354" then I'm never going to be quite sure who to trust. The easy way to avoid that is just to not pull stuff into the Entity, but leaving some sort of metadata trail might work too (just with more work).

I keep leaning more and more towards the most minimalist entities possible, unless we find some unavoidable reason to do more. (Eg maybe zoo animal entities will need to carry more data for whatever reasons.)

**Comment by campmlc** [comment]
If the Entity display won't be able to record a summary list of asserted
identifiers and their associated metadata, then I need some way of
downloading all asserted identifiers from the list in a way that can be
useable, e.g. the "parts/table download" view we discussed. The FLAT
display isn't sufficient. I can use the existing parts/table download to
get at parts.
I still think having the associated identifiers summarized on the Entity
page would be an important and valuable feature - but only if we can
implement the ID metadata so I can say "yes, I on this date accept this
assertion" or "No, I on this date do not accept this assertion".

**Comment by ccicero** [comment]
> Not completely Entity-related, but your data currently assert that you killed the same bluebird twice. Unless there's CPR involved, you need a better [event type](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecimen_event_type).

Ah, thanks. I changed those two records to 'encounter.' Is there a way to batch update the event type for that accession, or can you do it for me (accession MVZ:Bird 15700).

> At some level I don't care, accessions are administrative and you can do whatever you want with them, but I think accessions dedicated to entities is probably more correct/less confusing. You didn't actually receive anything for Entities, they are by definition not THINGS, you magicked them out of other data, keeping that separated seems worthwhile.

I agree with that - but your 'Create an Entity' tool from cloning a record doesn't do that. So you'd need to create the accession and then somehow select the accession when you clone the record.

> 
> My current inclination is to not attach anything that you don't absolutely have to to the Entities, which will keep them out of most search results. MAYBE that will evolve as we get deeper into this, but if we get there it will be easier to add than to subtract.

If Arctos:Entity is in the dropdown list and you search that specifically, then you will get search results but without any information. That's not useful. But I get that you want to keep them out of most search results. So do we want it in the dropdown list at all? Or if we do, then why not include them in the search results? If I want all bluebirds from Madera County, why not also find the entities? This gets back to the question of what we're trying to accomplish. If we want the data to be discoverable, then we need some basic data (takes me back to 'who, what, where, when').

> 
> We could test excluding them from search results by default, but that's going to have some computational cost; I'm not sure it's worth tackling that until there's some actual reason (eg clearly confused user) to do so.

> 
> I could pull whatever data into Entities, but it will get stale (a collector will contact you with a fix, you'll update the "real" catalog record and skip the Entity) so if we do go there I'd like to have some mechanism to tell what's been added by scripts and what might have been added by people for whatever reason. (The current 'entity magic' tab is 'by people,' it just saves you some clicks.)

That makes sense.
> 
> The 4th at 10 works for me.

I'll plan on it.

**Comment by Jegelewicz** [comment]
I sent everyone a calendar invite for the 4th at 10 PDT

**Comment by dustymc** [comment]
>do it for me

Confirm this is them:
```
create table temp_ccenc as
select guid,specimen_event.specimen_event_type,specimen_event.specimen_event_id
from
flat
inner join specimen_event on flat.collection_object_id=specimen_event.collection_object_id
inner join cataloged_item on flat.collection_object_id=cataloged_item.collection_object_id
inner join accn on cataloged_item.accn_id=accn.transaction_id
inner join collection on accn.collection_id=collection.collection_id
where collection.guid_prefix='MVZ:Bird' and accn_number='15700';
```
[temp_ccenc.csv.zip](https://github.com/ArctosDB/arctos/files/8393570/temp_ccenc.csv.zip)

>do we want it in the dropdown list at all?

I think it's too early to tell - i can't quite tell what @campmlc is thinking with the IDs, maybe that'll end up in some way useful to search. With these data, I think the only purpose the entity serves is to glue "components" together and I can't see why anyone would want to find them except maybe after having found the components. (Lack of imagination, probably!)

>want all bluebirds from Madera County..why not also find the entities

I think the question is why you want them. If youre doing something about individuals you probably only want entities or records that do not have an OrganismID. If you're looking at individual changes over time you only want records with multiple encounters (and maybe a collection) stitched together by entities. If you're looking at population genetics you want to avoid all but one of the "components" but otherwise don't care about the entities. I'm probably wrong about all of those things, and there are probably hundreds of use cases I haven't imagined. "Keep it simple" still seems like the best way to not eliminate any possibilities.

After there's some data, maybe someone (ideally someone with some entity-involved actual use case) should try to put together a demo/webinar, see can't be done, and we can try to make whatever that turns out to be somehow possible.

**Comment by ccicero** [comment]
@dustymc yes, those are the records for MVZ:Bird accession 15700 that need to be changed from specimen_event_type = 'collection' to 'encounter'

Thanks!

**Comment by dustymc** [comment]
add:
* ignore entities checkbox default checked on specimensearch
* some sort of text in specimenresults - idea is ~"this is an entity record it might not be what you expect" @ccicero to provide text

**Comment by Jegelewicz** [comment]
Carla’s Questions today
1. Entities need their own accession - but cloning adds them to the original accession.
  - Dusty will mess around to see if we can create an accession while cloning or add to accession while cloning
  - Also useful for parasite/hosts cloned from each other (belong in different collections with different accessions)
  - They need to look special in search results
  - Dusty suggests adding static text to indicate what they are “Click GUID to view components”
2. They are confusing in search results
  - Include check box “include entities” but default is no
  - Mariel thinks they should be visible because people won’t know what checking/unchecking the box means/does
  - Dusty - maybe we need more than one UI depending upon use - Mariel pretty much agrees
“  - Your results include entities - do you want to include entities? Checkbox, remove entities
3. Can we see something in related items that shows all components?
  - Once we figure out other id metadata
  - Need new relationships (component of, has component)
  - Magic in all related ids?
4. Could we see all components from multiple entities on a map?
  - Get your entities
  - Need a path to get all components of those entities
5. We need an entity creation tool (use a bird band or other unique “Organism Id” to create an entity and associate all components.
  - Dusty likes and is going to work on this

**Comment by Jegelewicz** [comment]
We could also register our entity domain on http://bioguid.org

**Comment by ccicero** [comment]
> 1. Entities need their own accession - but cloning adds them to the original accession.

FYI - I just fixed the one bluebird record so it's now in Arctos:Entity accession 4. I'll add the total # records once I figure that out.

> 2. They are confusing in search results
> 
> * Include check box “include entities” but default is no
> * Mariel thinks they should be visible because people won’t know what checking/unchecking the box means/does
> * Dusty - maybe we need more than one UI depending upon use - Mariel pretty much agrees
>   “  - Your results include entities - do you want to include entities? Checkbox, remove entities

Update: instead of check box on Search page, we decided to include them in search but have something on the results page that says something like "Your results include entity records [with link to documentation for what an entity is]. Check box to remove from results set." [where with one click you can remove all entity records]> 

> 5. We need an entity creation tool (use a bird band or other unique “Organism Id” to create an entity and associate all components.

This would be AWESOME! I added the bird band number to Arctos:Entity:33 and will wait for Dusty's magic tool before creating new entity records for the remaining bluebirds.

**Comment by dustymc** [comment]
>http://bioguid.org/

Excepting social problems like https://github.com/ArctosDB/arctos/issues/4200, Arctos identifiers are generally born actionable and linked via "how the internet works" - what could an additional registry DO for us?

@ccicero see https://github.com/ArctosDB/arctos/issues/3765#issuecomment-1087814255 - some sort of data in test would be very useful (maybe necessary) for this, do you happen to have the record bulkloader hanging around?

**Comment by Jegelewicz** [comment]
> what could an additional registry DO for us?

People are terrible at using https://arctos.database.museum/guid/Arctos:Entity:13 BUT if they had Arctos:Entity:3 and plugged it into BioGUID (and we had registered our domain there), they could find that they are missing the https://arctos.database.museum/guid/ part of an actionable thing.

BUT it does bring up a good point, because ALL colelctions registered there from Arctos would use https://arctos.database.museum/guid/ as their "dereference service prefix" and if someone only as MSB Mamm 5000, they will never get where they need to be.

Just for grins

I put MSB:Mamm:5000 into the BioGUID search, and I got the GBIF occurrence record
![image](https://user-images.githubusercontent.com/5725767/161631363-37352255-cad3-44e6-a5c2-b6b4ce053b26.png)

If you Google MSB:Mamm:5000 you get some GBIF and iDigBio stuff
![image](https://user-images.githubusercontent.com/5725767/161631791-9891d41c-b91e-4d9f-8508-9c0289e8873e.png)

Can we make it so that searching MSB:Mamm:5000 gets the Arctos record into the search results?

**Comment by dustymc** [comment]
>People are terrible at

Because we don't and that leaks, see https://github.com/ArctosDB/arctos/issues/4200 (and loan instructions and etc etc etc.) This is a social problem and those seldom have satisfactory technical solutions. We're currently demanding cruddy data, and that's what we get.

>Arctos:Entity:13

That's probably on the "computers might actually figure it out" end of the spectrum, but it's still a string - anyone can create or use it for any reason, any number of times, and then you can never be truly sure they think it's what you think it might be. We have good identifiers, we just refuse to use them.

>Can we make it so that

Given enough resources maybe we could get Google to do better, but the real answer will always revolve around Google doing whatever Google wants to do.

So, maybe bioguid could be useful, but only if we refuse to embrace real IDs (and who is in that group? Hopefully nobody!).

ANYWAY - there's magic in next release, should go out in an hour or so, go break it at test before making a giant mess in production please! I poked around at some [elephants](http://test.arctos.database.museum/guid/Arctos:Entity:13) and a [mouse](http://test.arctos.database.museum/guid/Arctos:Entity:15) and can't break it.

**Comment by dustymc** [comment]
>Your results include entity records [with link to documentation for what an entity is]. Check box to remove from results set." [where with one click you can remove all entity records]>

Done-ish, but the current documentation isn't great so no link.

**Comment by Jegelewicz** [comment]
Magic

Looks cool and a great way to boot your identification stats! :-)

![image](https://user-images.githubusercontent.com/5725767/161780635-917e872e-1853-4d75-8ede-314fd035d230.png)

Any way we can check for dupes an keep people from making messes?

**Comment by ccicero** [comment]
@dustymc I won't be able to work more on this until after my vacation (gone 4/7-4/16). What do you need from me? I have the bulkloader from the eagle data, but haven't done it yet for the bluebirds. Do you need me to send you the eagle bulkloader?

**Comment by dustymc** [comment]
@ccicero I found some elephants to play with, I think the tool is happy, have fun!

**Comment by campmlc** [comment]
Agree

**Comment by dustymc** [comment]
Add to entity magic form: 
* option to exclude singletons
* exclude records that already have OrganismID (or "pick an ID")?? Looks like this is going to happen in small batches, build with that in mind.

**Comment by dustymc** [comment]
@ccicero I snuck results/manage/entity magic/requery into prod.

**Comment by ccicero** [comment]
@dustymc cool, thanks! I'll start playing around.

Meanwhile, here is an entity record for an owl where we have the skeleton and AMNH has the skin:
https://arctos.database.museum/guid/Arctos:Entity:33
with component
https://arctos.database.museum/guid/MVZ:Bird:160626

I re-purposed that entity # from a bluebird because I want all the bluebirds to be consecutive.

All good (I think but please check) except for the map. There are no coordinates in the component record, but the map is showing a point off the west coast of Africa. The locality is in Thailand. ????
![image](https://user-images.githubusercontent.com/967275/165179152-42abe41d-0a41-4dfc-9bcf-91153bad0b51.png)

**Comment by dustymc** [comment]
>no coordinates

https://github.com/ArctosDB/arctos/issues/4546#issuecomment-1104057656

A georeference is two clicks away - click this:

<img width="428" alt="Screen Shot 2022-04-25 at 2 43 56 PM" src="https://user-images.githubusercontent.com/5720791/165180182-8690b3cc-830d-4d4b-be66-1609b918d70b.png">

then save (or I can do that for your collection, or some subset, or whatever).

**Comment by campmlc** [comment]
This mapping occurs when the entity has no event. See Entity 1

**Comment by ccicero** [comment]
@campmlc ok, but that shouldn't be the case.

@dustymc I can magic click the coordinates, and was going to do that but wanted to show you first. Seems like there should be no map if no coordinates?

**Comment by dustymc** [comment]
>This mapping occurs when the entity has no event. 

This is true only when the entity is a component of itself, which should probably never be the case (but whatever, I can't and won't try to stop such things, maybe there are good reasons to build self-referencing Entities).

>should be no map if no coordinates

Maybe, but I've come around to the idea that a (0,0) point is more informative than a big empty map. If we're going to change that then maybe we need to entirely rethink the prominent map (which I like!).

**Comment by campmlc** [comment]
Agree

**Comment by ccicero** [comment]
Mapping: A topic for discussion at our next organism meeting on May 9th.

I do like the big prominent map, but if there are no coordinates, they why have a map at all? We don't for records without coordinates.

**Comment by dustymc** [comment]
This has become unwieldy (I hate this thing: <img width="224" alt="Screen Shot 2022-05-03 at 8 59 45 AM" src="https://user-images.githubusercontent.com/5720791/166490505-62c05c66-438e-4d25-9d04-5e1d4502a5be.png">)  I opened a new issue for what I think is the only remaining discussion, closing.
